kombu/transport/SQS.py

```diff
@@ -151,7 +151,7 @@ def _reset_cycle(self):
     def entity_name(self, name, table=CHARS_REPLACE_TABLE):
         """Format AMQP queue name into a legal SQS queue name."""
         if name.endswith('.fifo'):
-            partial = name.rstrip('.fifo')
+            partial = name[:-len('.fifo')]
```

Comment:
Why is this implementation better?
What was wrong with rstrip that you have to change it?

Reply:
Nevermind. I didn't read the PR description.

Reply:
`rstrip` is a trap. Not the first time I see someone stumble into it.